code/game/objects/items/stacks/medical.dm

```diff
@@ -41,92 +47,103 @@
 		return TRUE
 
 	H.UpdateDamageIcon()
+
+	if(user.skills.getRating("medical") < skill_level_needed)
+		if(user.do_actions)
+			to_chat(user, span_warning("You're busy with something else right now!"))
+		if(!do_mob(user, M, unskilled_delay, BUSY_ICON_UNSKILLED, BUSY_ICON_MEDICAL))
+			return TRUE
+
 	return affecting
 
-/obj/item/stack/medical/bruise_pack
-	name = "roll of gauze"
-	singular_name = "medical gauze"
-	desc = "Some sterile gauze to wrap around bloody stumps."
-	icon_state = "brutepack"
-	heal_brute = 1
+/obj/item/stack/medical/heal_pack
+	name = "platonic gauze"
+	///How much brute damage this pack heals when applied to a limb
+	var/heal_brute = 0
+	///How much burn damage this pack heals when applied to a limb
+	var/heal_burn = 0
+	///Set of wound flags applied by use, including BANDAGE, SALVE, and DISINFECT
+	var/heal_flags = NONE
 
 
-/obj/item/stack/medical/bruise_pack/attack(mob/living/carbon/M as mob, mob/user as mob)
+/obj/item/stack/medical/heal_pack/attack(mob/living/carbon/M as mob, mob/user as mob)
 	. = ..()
 	if(. == TRUE)
-		return TRUE
-
-	if (ishuman(M))
-		var/mob/living/carbon/human/H = M
+		return
 
-		if(user.skills.getRating("medical") < SKILL_MEDICAL_PRACTICED && !do_mob(user, M, 1 SECONDS, BUSY_ICON_UNSKILLED, BUSY_ICON_MEDICAL))
-			return TRUE
+	var/datum/limb/affecting = .
+	var/mob/living/carbon/human/patient = M //If we've got to this point, the parent proc already checked they're human
+	if(affecting.surgery_open_stage) //Checks if mob is lying down on table for surgery
+		if(patient.can_be_operated_on())
+			do_surgery(patient, user, src)
+		else
+			to_chat(user, span_notice("\The [affecting.display_name] is cut open, you'll need more than a bandage!"))
+		return
 
-		var/datum/limb/affecting = .
+	var/affected = FALSE
+	if(heal_flags & BANDAGE)
+		if(affecting.bandage())
+			affected = TRUE
+	if(heal_flags & SALVE)
+		if(affecting.salve())
+			affected = TRUE
+	if(heal_flags & DISINFECT)
+		if(affecting.disinfect())
+			affected = TRUE
```

Comment:
all of this could be simplified .
for exemple, affected = (heal_flags & BANDAGE && affecting.bandage()) || ... the rest

Reply:
I'd like to keep the blocks separate for legibility, but yeah should probably affected |= affecting.disinfect() at least.